ProjFS.Mac/PrjFSKextTests/KextMockUtilities.hpp

```diff
@@ -120,21 +139,40 @@ bool ArgumentsAreEqual(TUPLE_T& arguments, CHECK_TUPLE_T check, std::index_seque
     return (ArgumentIsEqual(std::get<INDICES>(arguments), std::get<INDICES>(check)) && ...);
 }
 
-
 template <typename R, typename... ARGS>
 template <typename... CHECK_ARGS>
-    bool SpecificFunctionCallRecorder<R, ARGS...>::DidCallFunction(FunctionPointerType function, CHECK_ARGS... checkArgs)
+    bool SpecificFunctionCallRecorder<R, ARGS...>::DidCallFunction(FunctionPointerType function, int sequenceNumber, CHECK_ARGS... checkArgs)
 {
     std::pair<typename RecordedCallMap::const_iterator, typename RecordedCallMap::const_iterator> foundCalls = this->recordedCalls.equal_range(function);
     for (typename RecordedCallMap::const_iterator foundCall = foundCalls.first; foundCall != foundCalls.second; ++foundCall)
     {
         if (foundCall->second.argumentValues)
         {
-            if (ArgumentsAreEqual(*foundCall->second.argumentValues, std::forward_as_tuple(checkArgs...), std::index_sequence_for<CHECK_ARGS...>{}))
+            if (ArgumentsAreEqual(*foundCall->second.argumentValues, std::forward_as_tuple(checkArgs...), std::index_sequence_for<CHECK_ARGS...>{}) &&
+                (sequenceNumber == -1 || foundCall->second.callSequenceNumber == sequenceNumber))
             {
                 return true;
             }
         }
     }
     return false;
 }
+
+template <typename R, typename... ARGS>
+template <typename... CHECK_ARGS>
+    bool SpecificFunctionCallRecorder<R, ARGS...>::DidCallFunction(FunctionPointerType function, CHECK_ARGS... checkArgs)
+{
+    return DidCallFunction(function, -1, checkArgs...);
+}
+
+template <typename R, typename... ARGS>
+int SpecificFunctionCallRecorder<R, ARGS...>::CallCount(FunctionPointerType function)
+{
+    int callCount = 0;
+    std::pair<typename RecordedCallMap::const_iterator, typename RecordedCallMap::const_iterator> foundCalls = this->recordedCalls.equal_range(function);
+    for (typename RecordedCallMap::const_iterator foundCall = foundCalls.first; foundCall != foundCalls.second; ++foundCall)
```

Comment:
Rather than writing the loop yourself you can use [std::distance](http://www.cplusplus.com/reference/iterator/distance/) to get the count:

```
return distance(foundCalls.first, foundCalls.second);
```